dask/base.py

```diff
@@ -141,7 +142,14 @@ def normalize_function(func):
         kws = tuple(sorted(func.keywords.items())) if func.keywords else ()
         return (normalize_function(func.func), func.args, kws)
     else:
-        return str(func)
+        try:
+            return pickle.dumps(func, protocol=0)
+        except:
+            try:
+                import cloudpickle
+                return cloudpickle.dumps(func, protocol=0)
+            except:
+                return str(func)
```

Comment:
Conversations with @maxhutch made it clear that functions weren't being tokenized consistently across Python sessions

```python
In [1]: from toolz import identity

In [2]: str(identity)
Out[2]: '<function identity at 0x7fac72f4a598>'
```